homework/index.js

```diff
@@ -15,32 +38,127 @@
     xhr.onerror = () => cb(new Error('Network request failed'));
     xhr.send();
   }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
+  // *********************************************************
+  function ShowContributions(urlText) {
+    const infoContainer = document.getElementById('info_container');
+    const infoRight = createAndAppend('div', infoContainer, {
+      class: 'info-right',
+      text: 'Contributions:',
+    });
+    fetchJSON(urlText, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', infoRight, {
+          class: 'contributor-link',
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+          width: '50px',
+        });
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
     });
-    return elem;
   }
-
-  function main(url) {
+  // *************************************************
+  function ShowRepoInfo(arrayOfObjects, indexNumber) {
+    const wantedRepo = arrayOfObjects[indexNumber];
+    const infoContainer = document.getElementById('info_container');
+    infoContainer.innerHTML = '';
+    const infoLeft = createAndAppend('div', infoContainer, {
```

Comment:
An HTML table would provide a nicer layout here.

Reply:
I am going to try that. I am reading the code in JavaScript 3 Examples. 
I am still having problem there with <a> tag. 
Actually just now (while I am writing here) I got an idee to solve it. 
I think it is going to work, otherwise I will ask again about it.